src/ucp/api/ucp_def.h

```diff
@@ -81,6 +82,29 @@ typedef struct ucp_config                ucp_config_t;
 typedef struct ucp_ep                    *ucp_ep_h;
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Callback to process incoming active message
+ *
+ * @param [in]  arg     User-defined argument.
+ * @param [in]  data    Points to the received data. 
+ * @param [in]  length  Length of data.
+ * @param [in]  flags   Ignore, these are only relevant at the UCT level
+ *
+ * @note This callback could be set and released
+ *       by @ref ucp_worker_set_am_handler function.
+ *       The minimum value of length is 8. If a user only
```

Comment:
8 of what units ?

Reply:
The minimum value that will be passed into the length parameter of the callback will be 8 bytes because a header for a uct short message is of type uint64_t, and the header is always sent. I wanted users to be aware that if they, for example, send 1 byte of data, length will still be set to 8 when the callback is invoked.

Reply:
but the header is not exposed to the user ?
You can do +8 internally, right ?

Reply:
The header is exposed to the user. data includes the header. The header only exists for short messages and the invocation of the callback does not change between short/bcopy/zcopy, so when doing a short message, we just use the header to store 8 bytes of the data. It works very similarly to 
https://github.com/openucx/ucx/blob/master/test/examples/uct_hello_world.c#L89